funcbench/main.go

```diff
@@ -0,0 +1,362 @@
+		}
+
+		if err := gitClient.fetchPullRequest(); err != nil {
+			log.Fatalln(err)
+		}
+		if err := gitClient.checkoutToPullRequest(); err != nil {
+			comment := fmt.Sprintf("Switch to a pull request branch failed. %s", ghClient.logLink)
+			if postCommentErr := ghClient.postComment(comment); postCommentErr != nil {
+				log.Fatalf("An error: %v occured while processing error: %v", postCommentErr, err)
+			}
+			log.Fatalln(err)
+		}
+
+		out, err := benchTest.execBenchmark()
+		if err != nil {
+			comment := fmt.Sprintf("Go bench test for this pull request failed. %s", ghClient.logLink)
+			if postCommentErr := ghClient.postComment(comment); postCommentErr != nil {
+				log.Fatalf("An error: %v occured while processing error: %v", postCommentErr, err)
+			}
+			log.Fatalln(err)
+		}
+		if err := ioutil.WriteFile(benchTest.pullRequestFilePath, out, 0644); err != nil {
+			log.Fatalln(err)
+		}
+
+		if err := gitClient.checkoutToComparisonBranch(); err != nil {
+			log.Fatalln(err)
+		}
+
+		out, err = benchTest.execBenchmark()
+		if err != nil {
+			comment := fmt.Sprintf("Go bench test for this pull request failed. %s", ghClient.logLink)
+			if postCommentErr := ghClient.postComment(comment); postCommentErr != nil {
+				log.Fatalf("An error: %v occured while processing error: %v", postCommentErr, err)
+			}
+			log.Fatalln(err)
+		}
+		if err := ioutil.WriteFile(benchTest.comparisonBranchFilePath, out, 0644); err != nil {
+			log.Fatalln(err)
+		}
+
+		unformatedComparisonTable, err := benchTest.compareBenchmarks()
+		if err != nil {
+			comment := fmt.Sprintf("Error: `benchcmp` failed. %s", ghClient.logLink)
+			if postCommentErr := ghClient.postComment(comment); postCommentErr != nil {
+				log.Fatalf("An error: %v occured while processing error: %v", postCommentErr, err)
+			}
+			log.Fatalln(err)
+		}
+
+		if strings.Count(unformatedComparisonTable, "\n") < 2 { // Maybe delete as this error gets caught sooner
+			comment := fmt.Sprintf("Error: `go test` did not match any `BenchmarkXxx` functions. %s", ghClient.logLink)
+			if postCommentErr := ghClient.postComment(comment); postCommentErr != nil {
+				log.Fatalf("An error: %v occured while processing error: %v", postCommentErr, err)
+			}
+		}
+		if err := ghClient.postComment(formatFinalComment(unformatedComparisonTable)); err != nil {
+			log.Fatalln(err)
+		}
+
+	default:
+		log.Fatalln("only issue_comment event is supported")
+	}
+}
+
+func newGitHubClient(event *github.IssueCommentEvent) *gitHubClient {
+	owner := *event.GetRepo().Owner.Login
+	repo := *event.GetRepo().Name
+	latestCommitHash := os.Getenv("GITHUB_SHA")
+
+	return &gitHubClient{
+		client:           newClient(os.Getenv("GITHUB_TOKEN")),
+		owner:            owner,
+		repo:             repo,
+		prNumber:         *event.GetIssue().Number,
+		latestCommitHash: latestCommitHash,
+		logLink:          fmt.Sprintf("Full logs at: https://github.com/%s/%s/commit/%s/checks", owner, repo, latestCommitHash),
```

Comment:
don't think this should be part of the github client, can be just a variable in the main function